### PR TITLE
producer: support producer

### PR DESCRIPTION
**Pull request by kasperpihl**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ Split your reducer into micro reducers based on actions, and avoid having one la
 
 useMicroReducer uses the standard useReducer under the hook 🎉
 
+Ohh, and it supports Immer 🎂
```
Comment (cenobitedk) [inline README.md]: Cake 😋

File: README.md [hunk 2]
```diff
@@ -47,6 +49,18 @@ export default function App() {
...
+import { setProducer } from "react-micro-reducer";
+setProducer(produce);
+```
+
+Now the state in your all your micro reducers will be an immer draft state 💪
+
 ## Credits
 
-Thanks to [Maciej Sikora](https://stackoverflow.com/a/59002901/1168927) for helping with the type definitions! 🙌
+Thanks to [Maciej Sikora](https://stackoverflow.com/a/59002901/1168927) for helping with the type definitions, and to [jhass](https://github.com/cenobitedk) for helping with the naming/API! 🙌
```
Comment (cenobitedk) [inline README.md]: My pleasure 🙏

File: README.md [hunk 3]
```diff
@@ -47,6 +49,18 @@ export default function App() {
 }
 ```
 
+## Usage with Immer
+
+Somewhere globally like index.ts or App.ts
```
Comment (cenobitedk) [inline README.md]: Can it be scoped to single file as well, or is it purely global?

Comment (kasperpihl) [reply README.md]: Well, we could do an optional third argument to useMicroReducer?

Comment (kasperpihl) [reply README.md]: Maybe that's actually better. You might have to include immer more places, but on the other hand WYSIWYG and it's more clear that the state has been modified

Comment (cenobitedk) [reply README.md]: so it'd be either
```
// app.tsx
setImmer(produce);

// file.tsx
const [...] = useMicroReducer(actions, state);
```
or 
```
// file.tsx
const [...] = useMicroReducer(actions, state, producer);
```

Comment (cenobitedk) [reply README.md]: That's actually pretty cool. Then what if you already have a global producer?

Comment (kasperpihl) [reply README.md]: Local always wins

Comment (kasperpihl) [reply README.md]: But I'm thinking to not do the global version at all. Is it making more confusion? Imagine opening a file and seeing the state getting mutated, at least by explicitly passing it you can always see that something is modifying it.

Comment (cenobitedk) [reply README.md]: True. Makes good sense :)

Comment (kasperpihl) [reply README.md]: Check the code now. How do you like them Apples? :D

**Pull request merged**